src/components/UserInfo/UserInfo.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import './UserInfo.scss';
+
+import { Users } from '../../types/Users';
+
+type Props = {
+  user: Users | undefined;
```

Comment:
```suggestion
  user: Users;
```
Так як ти цей компонент рендериш тільки у випадку, коли є post.user, то user тут буде завжди (тобто ніколи не буде undefined, бо якщо на вищому рівні він undefined, то компонент навіть не зрендериться)
<img width="379" alt="image" src="https://user-images.githubusercontent.com/66570142/159455509-e21d68b6-26c3-406e-b50e-f1b3fe0d2a17.png">